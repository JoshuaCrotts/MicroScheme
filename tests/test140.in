;
; Iterative factorial using do loop.
;

(define (fact-iterative n)
	(do ((idx 1 (+ idx 1)) (product 1))
		((> idx n) product)
		(set! product (* product idx))))

(fact-iterative 170)
