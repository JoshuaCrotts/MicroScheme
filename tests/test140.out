7.257415615307994E306
